crates/store/re_grpc_client/src/redap/address.rs

```diff
@@ -35,12 +35,15 @@ pub enum ConnectionError {
     /// The given url is not a valid Rerun storage node URL.
     #[error("URL {url:?} should follow rerun://host:port/recording/12345 for recording or rerun://host:port/catalog for catalog")]
     InvalidAddress { url: String, msg: String },
+
+    #[error("Catalog URL {origin:?} cannot be loaded as a recording")]
+    CannotLoadCatalogAsRecording { origin: Origin },
```

Comment:
I think instead of the `Origin` we should pass in the entire recording URL (`String`).

Btw, we you're at it could you please remove `InvalidAddress`? We are not using that anymore. Is there a way to find unused variants in libraries?